jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -2149,6 +2150,19 @@ public AsyncContext startAsync(ServletRequest servletRequest, ServletResponse se
         return _async;
     }
 
+    protected String getRequestURI(ServletRequest servletRequest)
+    {
+        if (servletRequest instanceof HttpServletRequestWrapper)
+        {
+            return ((HttpServletRequestWrapper)servletRequest).getRequestURI();
+        }
+        if (servletRequest instanceof ServletRequestWrapper)
+        {
+            return getRequestURI(((ServletRequestWrapper)servletRequest).getRequest());
+        }
+        return ((HttpServletRequest)servletRequest).getRequestURI();
+    }
```

Comment:
The code logic here is duplicated with the one in the Server class.
I would prefer to see a `static HttpServletRequest unwrap()` in class `Request` and similar in class `Response`.